server/api/user/update/service.js

```diff
@@ -0,0 +1,146 @@
+import has from 'lodash/has';
+import isEmpty from 'lodash/isEmpty';
+import {
+  InvalidPrimaryEmailError,
+  DuplicateEmailAddressError,
+  DuplicateUsernameError,
+  InvalidUsernameError,
+} from '../../../constants/errors';
+import deleteUserPicture from '../deletePicture/service';
+import commonNames from '../../../utils/commonNames';
+
+async function updateUser({ db, storage }, user, nextProps) {
+  const UserModel = db.model('User');
+  const CredentialsModel = db.model('Credentials');
+
+  const nextUser = {};
+  const nextCredentials = {};
+
+  // decorate nextUser obj with emails
+  if (nextProps.emails) {
+    // ensure there is exactly 1 primary email
+    const primaryEmails = nextProps.emails.filter((email) => email.isPrimary);
+    if (primaryEmails.length < 1) {
+      throw new InvalidPrimaryEmailError('You must specify at least 1 primary email');
+    }
+    if (primaryEmails.length > 1) {
+      throw new InvalidPrimaryEmailError('User cannot have more than 1 primary emails');
+    }
+
+    // if requestor changes the previous primary email it also needs to be verified
+    const primaryEmail = primaryEmails[0];
+    const emailExisted = user.emails.find((email) => primaryEmail.address === email.address);
+    if (!emailExisted || !emailExisted.isVerified) {
+      throw new InvalidPrimaryEmailError(
+        `You need to verify ${primaryEmail.address} before marking it as primary`
+      );
+    }
+
+    // normalize emails
+    const normalizedEmails = nextProps.emails.map((email) => {
+      return {
+        ...email,
+        address: UserModel.normalizeEmailAddress(email.address),
+        isVerified: email.isVerified || false,
```

Comment:
`isVerified` property no longer supports a default value in the controller since a user updating their own profile can NOT set the isVerified property to _any_ value. Setting it as a default to be false, means that they could also send the `isVerified` as false for their verified email. And we don't want them to update the isVerified property at _any_ case. Please advise if this is correct thinking

Reply:
What you did here is correct I think.

It doesn't matter if a simple user specifies `isVerified` as false or undefined. They are one and the same. However, my suggestion is to NOT allow simple users to specify `isVerified` at all - to avoid potential confusion/misconception and simplify the source code.

Reply:
Yeah simple users are not allowed to send over any `isVerified` value. It will throw an error in the `isAllowed` helper function inside the controller. This needs a minor fix though. In the case that a simple user sent over an update, this LOC here will also set their new emails as `isVerified: false` since the value will be missing, and update the DB which is wrong. Will fix asap